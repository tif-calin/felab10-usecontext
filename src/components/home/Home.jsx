import React from 'react';
import { Link } from 'react-router-dom';
import Header from '../header/Header';
import APIs from '../../data/characterAPIs.json';

const Home = () => {
  return (<>
    <Header/>
    <main className="Home">
      What characters would you like to see?
      <ul>
        {Object.keys(APIs).map(key => <li key={key}>
          <Link to={`/list?api=${key}`}>
            {APIs[key].name}
          </Link> ({APIs[key].count})
        </li>)}
      </ul>
    </main>
  </>);
};

export default Home;
