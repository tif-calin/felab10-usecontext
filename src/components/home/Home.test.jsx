import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Home from './Home';
import { MemoryRouter } from 'react-router-dom';

describe('Home component', () => {

  afterEach(() => cleanup());

  test('renders Home', () => {
    const { asFragment } = render(<MemoryRouter>
      <Home/>
    </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
  
});
