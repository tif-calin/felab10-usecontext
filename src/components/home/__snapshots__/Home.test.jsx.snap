// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders Home 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <h1>
      Lists of Things
    </h1>
    <nav>
      <a
        href="/"
      >
        home
      </a>
    </nav>
  </header>
  <main
    class="Home"
  >
    What characters would you like to see?
    <ul>
      <li>
        <a
          href="/list/twinpeaks"
        >
          Twin Peaks
        </a>
         (74)
      </li>
      <li>
        <a
          href="/list/avatar"
        >
          Avatar: The Last Airbender
        </a>
         (497)
      </li>
      <li>
        <a
          href="/list/rickandmorty"
        >
          Rick and Morty
        </a>
         (671)
      </li>
      <li>
        <a
          href="/list/heyarnold"
        >
          Hey Arnold!
        </a>
         (218)
      </li>
      <li>
        <a
          href="/list/animalcrossing"
        >
          Animal Crossing
        </a>
         (391)
      </li>
    </ul>
  </main>
</DocumentFragment>
`;
